boards/arc/nsim/nsim_hs_smp_defconfig

```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_ARC=y
+CONFIG_CPU_ARCHS=y
+CONFIG_SOC_NSIM=y
```

Comment:
Do you need both SOC_NSIM and SOC_NSIM_SMP set to yes?
Seems odd, having two SoCs selected for a single board.

Reply:
Well, this is how we used to handle different flavors of nSIM boards: we have `SOC_NSIM` for this platform in general and then `SOC_NSIM_EM`, `SOC_NSIM_SEM`, `SOC_NSIM_HS` & now `SOC_NSIM_HS_SMP`.

If I remove `SOC_NSIM` from `boards/arc/nsim/nsim_hs_smp_defconfig` on configuration I'm getting:
> warning: the choice symbol SOC_NSIM_HS_SMP (defined at soc/arc/snps_nsim/Kconfig:28) was selected (set =y), but no symbol ended up as the choice selection. 

and the whole lot of misconfiguration messages.

So I would agree that `CONFIG_SOC_NSIM` might be implicitly selected by `CONFIG_SOC_NSIM_HS_SMP` but that will require rework of nSIM Kconfig infrastructure.

That said IMHO it's better to merge nSIM HS_SMP changes first and then do a clean-up.

Reply:
Let's leave it like this, indeed for other ARC boards this is the same.